internal/pkg/cli/env_init.go

```diff
@@ -290,7 +291,15 @@ func (o *initEnvOpts) Execute() error {
 	}
 
 	// 4. Start creating the CloudFormation stack for the environment.
-	if err := o.deployEnv(app, urls); err != nil {
+	if resources.S3Bucket == "" {
+		log.Errorf("Cannot find the S3 artifact bucket in %s region created by app. The S3 bucket is necessary for many future operations. For example, when you need addons to your services.", envRegion)
```

Comment:
nit: not sure if this was intended:
```suggestion
		log.Errorf("Cannot find the S3 artifact bucket in %s region created by app %s. The S3 bucket is necessary for many future operations. For example, when you need addons to your services.", envRegion, app)
```